BuildGuide.md

```diff
@@ -1,3 +1,5 @@
+TODO: is there any info on this file that needs to be on the new AppGenGuide before deleting this file?
```

Comment:
While I don't thin the information in this file is **necessary** for the new AppGenGuide, It has condensed information that may be valuable for less experienced users (like our team).

Reply:
Just a suggestion, I do believe that this file has a little bit more information on each of the files in Flux and plugins that won't look nice to have inside the appGenGuide. Maybe get rid of information about build and have this be a file on the explanation of the structure of a StarlingX app, and of course increment it

Reply:
@Danmcaires @tnpb-encora and I decided that the info is useful and needs to be somewhere, but not as past of a build guide. A documentation, maybe, or in the README.md of the App Generator, but on another section.